### DISCUSSION TITLE
How can i set varchar instead of nvarchar

### DISCUSSION THREAD

**Discussion by garsiv1932** [abpframework/abp#11148] [General]
I want to change the database string type from nvarchar to varchar?

Normally i use this funnction in my projects

            foreach (var entityType in builder.Model.GetEntityTypes())
            {
                // Use the entity name instead of the Context.DbSet<T> name
                // refs https://docs.microsoft.com/en-us/ef/core/modeling/entity-types?tabs=fluent-api#table-name
        
                IEnumerable<IMutableProperty> propeties = entityType.GetProperties();
                foreach (var prop in propeties)
                {
                    string typename = prop.ClrType.Name;
                    if (typename == "String" )
                    {
                        prop.SetColumnType("varchar(200)");
                    }
                }
            }

The problem is i dont want to change the length of the columns and if i dont use a fixed length in SetColumnType the columns that are max become varchar(1).

Wow can i fix that?

Thanks in advance

**Comment by maliming** [comment]
Maybe you can get length by `prop.GetMaxLength()`, 
This is not related to ABP.

**Reply by garsiv1932** [reply]
I know it is more a .NET problem as this is is a issue that i have with pure .NET code.
Only want to know if someone had a fix for that.

Thanks for the answer.
